user-management/domain/user/user.go

```diff
@@ -0,0 +1,84 @@
+package user
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+
+	cours "github.com/mradulrathore/user-management/domain/course"
+
+	validation "github.com/go-ozzo/ozzo-validation"
+)
+
+type User struct {
+	Name         string       `json:"fullname"`
+	Age          int          `json:"age"`
+	Address      string       `json:"address"`
+	RollNo       int          `json:"rollno"`
+	CoursesEnrol cours.Course `json:"coursesenrol"`
+}
+
+func New(name string, age int, address string, rollNo int, courseEnrol []string) (User, error) {
+	var user User
+	var err error
+
+	user.Name = name
+	user.Age = age
+	user.Address = address
+	user.RollNo = rollNo
+	user.CoursesEnrol, err = cours.New(courseEnrol)
+	if err != nil {
+		return User{}, err
+	}
+
+	if err = user.validate(); err != nil {
+		log.Println(err)
+		return User{}, err
+	}
+
+	return user, nil
+}
+
+func (user User) validate() error {
+	return validation.ValidateStruct(&user,
+		validation.Field(&user.Name, validation.Required),
+		validation.Field(&user.Age, validation.Required, validation.By(checkPositive)),
+		validation.Field(&user.Address, validation.Required),
+		validation.Field(&user.RollNo, validation.Required, validation.By(checkPositive)),
+		validation.Field(&user.CoursesEnrol, validation.Required),
+	)
```

Comment:
explore whether we can use json tags

Reply:
there is a different package (called validator.v8) for that[(link)](https://pkg.go.dev/gopkg.in/go-playground/validator.v8#section-readme)